From ec982e77b652466da048834373f4b1fc65789238 Mon Sep 17 00:00:00 2001
From: Morgane BUTSCHER <morganex.butscher@intel.com>
Date: Thu, 2 May 2013 12:28:48 +0200
Subject: [PORT FROM MAIN] ril: RIL 6.49 and 6.50 enhancement

BZ: 105068

Release Notes for M6.49
========================

1. Sync external files from WW14 ABSP:
-Update: /hardware/ril/include/telephony/ril.h
/libril/ril.cpp
/vendor/intel/clovertrail/board/redhookbay/rril/repository6360.txt
-Add: /vendor/intel/clovertrail/board/redhookbay/rril/repository6268.txt
/repository7160.txt

2. Bugzilla 97878: RIL shall support Preferred Network Type LTE_WCDMA
-Add new preferred AcT type, PREF_NET_TYPE_LTE_WCDMA (12) in ril.h
-Create command to set new mode (AT+XACT=4,2) in CTE_XMM7160::CoreSetPreferredNetworkType.

3. Bugzilla 97767: Post command handler should not be called for no-op command
-Add check for NULL command (NO-OP) before calling the Post command handler
in CChannel::SendCommand().

4. Bugzilla 86030: Clear ip addresses at interface down
-Integrate patch 96211 (already merged)

5. Bugzilla 95515: Add option 1 and 2 to the settings of rf power cutback
-Integrate patch 98193 (already merged)

6. Bugzilla 96444: Reset RadioRequestPending on radio/modem events
-Integrate patch 100230 (already merged)

7. Bugzilla 95346: Cache screen state value in radio off
-Integrate patch 100231 (already merged)

8. Bugzilla 90105: Handling request type in Deactivate data
-Integrate patch 97362 (already merged)

9. Bugzilla 94763: Correct segmentation fault on repository
-Integrate patch 99525 (already merged)

10. Bugzilla 96184: Fix network loss due to port opening failure
-Integrate patch 98986 (already merged)

11. Bugzilla 96001: Update channel assignment for call operations
-Integrate patch 99682 (already merged)

Release Notes for M6.50
========================

1. Bugzilla 88124: Ignore +XREG unsolicited when the only modification is band change
-In CSilo_Network::ParseRegistrationStatus function, compare cached XREG info and
current XREG info. If nothing has changed (state, AcT, LAC, or CID) or only the
band has changed, ignore notification (don't report to Android) by setting
SetUnrecognizedFlag(TRUE).

2. Bugzilla 93765: LTE RIL implementation cleanup
-Integrate patch 100522 (already merged)

3. Bugzilla 97386: LTE RIL, non LTE data connection can be established only one time
after DUT reboot
-Integrate patch 100523 (already merged)

4. Bugzilla 95201: Adding missing Telephony APIs for Lenovo
-Integrate patch 102704 (already merged)

5. Bugzilla 94183: Update the response pointer after parsing of XDRVI
-Integrate patch 102851 (already merged)

6. Bugzilla 96043: Fix first DTMF request ignored on some cases
-Integrate patch 102852 (already merged)

7. Bugzilla 98152: Fix rild socket closed due to modem events
-Integrate patch 102853 (already merged)

8. Bugzilla 93807: Fix file descriptor leak in DataConfigDown
-Integrate patch 104682 (already merged)

9. Bugzilla 103189: Don't destroy system manager on modem events
-Integrate patch 104521 (already merged)

10. Bugzilla 94202: Revert "Modifictations to support LTE registration"
-Integrate patch 102210 (already merged)

11. Bugzilla 95201: Add handling of missing Telephony Enabler APIs
-Integrate patch 102711 (already merged)

12. Bugzilla 95201: Adding missing API of PIN retry count for Lenovo
-Integrate patch 104371 (already merged)

Change-Id: I7899015993f2e1eb23ad8a888dfa8ad996830ebc
Orig-Change-Id: Ib21e7299ebd08f6c6179c55eaebed10c4f7438b0
Signed-off-by: Morgane BUTSCHER <morganex.butscher@intel.com>
Reviewed-on: http://android.intel.com:8080/105773
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Lucas, Guillaume <guillaume.lucas@intel.com>
Reviewed-by: Marmounier, Edward <edward.marmounier@intel.com>
Reviewed-by: Fremaux, Pascal <pascal.fremaux@intel.com>
Reviewed-by: Roques, CyrilX <cyrilx.roques@intel.com>
Reviewed-by: Ravindran, Arun <arun.ravindran@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Badrappan, Jeevaka <jeevaka.badrappan@intel.com>
Tested-by: Louis, NicolasX <nicolasx.louis@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 include/telephony/ril.h |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/include/telephony/ril.h b/include/telephony/ril.h
index b528312..a8f5a08 100644
--- a/include/telephony/ril.h
+++ b/include/telephony/ril.h
@@ -128,7 +128,8 @@ typedef enum {
     PREF_NET_TYPE_LTE_CDMA_EVDO            = 8, /* LTE, CDMA and EvDo */
     PREF_NET_TYPE_LTE_GSM_WCDMA            = 9, /* LTE, GSM/WCDMA */
     PREF_NET_TYPE_LTE_CMDA_EVDO_GSM_WCDMA  = 10, /* LTE, CDMA, EvDo, GSM/WCDMA */
-    PREF_NET_TYPE_LTE_ONLY                 = 11  /* LTE only */
+    PREF_NET_TYPE_LTE_ONLY                 = 11, /* LTE only */
+    PREF_NET_TYPE_LTE_WCDMA                = 12  /* LTE/WCDMA (LTE preferred) */
 } RIL_PreferredNetworkType;
 
 /* Source for cdma subscription */
-- 
1.7.4.1

